backend/api/quivr_api/modules/sync/utils/list_files.py

```diff
@@ -292,3 +295,127 @@ def get_azure_files_by_id(credentials: dict, file_ids: List[str]):
         file.name = remove_special_characters(file.name)
     logger.info("Azure Drive files retrieved successfully: %s", len(files))
     return files
+
+
+# Drop Box
+def list_dropbox_files(
+    credentials: dict, folder_id: str = "", recursive: bool = False
+) -> List[SyncFile] | dict:
+    """
+    Retrieve files from Dropbox.
+
+    Args:
+        credentials (dict): The credentials for accessing Dropbox.
+        folder_id (str, optional): The folder ID to filter files. Defaults to "".
+        recursive (bool, optional): If True, fetch files from all subfolders. Defaults to False.
+
+    Returns:
+        dict: A dictionary containing the list of files or an error message.
+    """
+    logger.info("Retrieving Dropbox files with folder_id: %s", folder_id)
+
+    # Verify credential has the access token
+    if "access_token" not in credentials:
+        raise HTTPException(status_code=401, detail="Invalid token data")
```

Comment:
HTTPException should only appear at the HTTP layer (routes) of the app. This function should raise some Exception (to define) and catched on the layer above